substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrBufferAccess.java

```diff
@@ -91,19 +120,35 @@ public static boolean isLocked(JfrBuffer buffer) {
     @Uninterruptible(reason = "We must guarantee that all buffers are in unacquired state when entering a safepoint.", callerMustBe = true)
     public static boolean tryLock(JfrBuffer buffer) {
         assert buffer.isNonNull();
-        return NativeSpinLockUtils.tryLock(ptrToLock(buffer));
+        boolean result = NativeSpinLockUtils.tryLock(ptrToLock(buffer));
```

Comment:
I added [`JavaOwnedSpinLockUtils`](substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/JavaOwnedSpinLockUtils.java) that is similar to `JavaSpinLockUtils` but does CAS with the acquiring thread. I thought it might be cleaner than adding new methods to `JavaSpinLockUtils`. What do you think?